### PR TITLE
Fix/1795 input number safari

### LINKED ISSUE #1795: Input - "Number only" actually accepts other characters in Safari
**Describe the bug**
Input - "Number only" actually accepts other characters in Safari

**Steps to reproduce**
1. Go to [Input demo](https://sui-components.vercel.app/workbench/atom/input/demo) in Safari
2. Scroll down to the Number example
3. See error

https://user-images.githubusercontent.com/23620759/136557581-799a31a8-b253-4483-b505-04925d61b76b.mov

**Expected behaviour**
This option should only allow numbers

**Environment / Context**
- Desktop or Mobile
- Browser and version Safari 14.1.2
- Device: Macbook

### PR DESCRIPTION
**Pull request by stefi23**
## Category/Component

**TASK**:  #1795 

### Types of changes

- [x] 🪲 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🧾 Documentation
- [ ] 📷 Demo
- [ ] 🧪 Test
- [ ] 🧠 Refactor
- [ ] 💄 Styles

### Description, Motivation and Context
- Input type number accepts other characters in the Safari and Firefox browser
- To fix the issue an additional validator function was added to the input type number

### CODE REVIEW

File: components/atom/input/src/index.js [hunk 1]
```diff
@@ -8,11 +8,29 @@ import Password from './Password/index.js'
...
+  }
   switch (type) {
     case 'sui-password':
       return <Password ref={ref} {...props} />
     case 'mask':
       return <Mask ref={ref} {...props} />
+    case 'number':
+      return (
+        <Input
+          onKeyDown={e => checkIfValidNumberInput(e)}
```
Comment (andresin87) [inline components/atom/input/src/index.js]:
```suggestion
          onKeyDown={checkIfValidNumberInput}
```

File: components/atom/input/src/index.js [hunk 2]
```diff
@@ -8,11 +8,29 @@ import Password from './Password/index.js'
 import {INPUT_SHAPES, TYPES} from './config.js'
 
 const AtomInput = forwardRef(({type, ...props}, ref) => {
+  const checkIfValidNumberInput = event => {
```
Comment (andresin87) [inline components/atom/input/src/index.js]: move it to settings and easy import it to not redeclare the function on every declaration hoisting and simply get the outside defined

Comment (stefi23) [reply components/atom/input/src/index.js]: Thank you! I moved the function to a helper.js file. Would it be preferred to have it in the config file or in a setting named file?

Comment (andresin87) [reply components/atom/input/src/index.js]: =)

File: components/atom/input/src/index.js (hunk 1 above)
Comment (andresin87) [reply components/atom/input/src/index.js]: simply direct set the fn to not declare a new fn handler on every render

**Pull request merged**